### DISCUSSION TITLE
Provide credentials to post upgrade task

### DISCUSSION THREAD

**Discussion by MShekow** [renovatebot/renovate#22988] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.142.0

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Suppose I want to run a post upgrade command such as `helm registry login <host> -u <username> -p <password>` where the credentials are already configured as **host rules**. How do I do that?

It seems that the environment variables that are configured when running post upgrade tasks are excluding those that I set when running the Renovate image. I run Renovate inside the `renovate/renovate` Docker image.

### What you tried so far.

Nothing, as I have no idea what to do. A super-ugly hack (that I did not try yet) might be that I make the credentials available via files:

In `config.js` I could add this:

```js
const fs = require('fs');

fs.writeFile('/tmp/credentials', process.env.REGISTRY_PASSWORD, err => {
  if (err) {
    console.error(err);
  }
});
```
and in my post upgrade command I could read the password from `/tmp/credentials`.

**Comment by rarkins** [comment, marked as answer]
Try putting the credentials into `secrets` and then templating them on your command

**Reply by MShekow** [reply]
You mean https://docs.renovatebot.com/self-hosted-configuration/#secrets ? I'll try, thank you.

**Reply by rarkins** [reply]
Yes, so do this:
- Add credentials in `secrets` (global config)
- Enable `allowPostUpgradeCommandTemplating` (global config)
- Reference `{{ secrets.X }}` in `hostRules` and `postUpgradeTasks` (can be in global config or repo config)

**Reply by MShekow** [reply]
Your suggestion worked, thank you. I think that I did not discover it myself, because https://docs.renovatebot.com/templates/ does not mention the `secrets`.
